src/SiteMaster/Core/Util.php

```diff
@@ -233,14 +233,23 @@ public static function getHTTPInfo($url, $options = array())
         
         curl_exec($curl);
 
-        $http_status = curl_getinfo($curl, CURLINFO_HTTP_CODE);
+        $http_status   = curl_getinfo($curl, CURLINFO_HTTP_CODE);
         $effective_url = curl_getinfo($curl, CURLINFO_EFFECTIVE_URL);
         $curl_error_no = curl_errno($curl);
 
         curl_close($curl);
         
+        //Always sleep after requesting data, this will help prevent loops from flooding a server.
+        sleep(1);
+        
         $okay = self::httpCodeIsOkay($http_status);
 
+        //Sometimes HEAD requests result in a false negative.  Do a GET request just to verify
+        if (true === $options[CURLOPT_NOBODY] && false === $okay) {
```

Comment:
I think it would be better if we had a stronger heuristic for false positives. With this addition, every 404 page will end up getting hit twice. I understand if there just isn't a way to avoid this given the reports of issues.